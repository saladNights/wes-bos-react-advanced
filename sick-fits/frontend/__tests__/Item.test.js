import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';

const fakeItem = {
	id: '234rf2t',
	title: 'An Item',
	price: 5000,
	description: 'This is an item description',
	image: 'dog.jpg',
	largeImage: 'dogXL.jpg',
};

describe('Component', () => {
	it('renders properly',  () => {
		const wrapper = shallow(<ItemComponent item={fakeItem} />);
		const PriceTag = wrapper.find('ForwardRef(PriceTag)');

		expect(PriceTag.children().text()).toBe('$50');
		expect(wrapper.find('ForwardRef(Title) Link a').text()).toBe(fakeItem.title);

		const img = wrapper.find('img');

		expect(img.props().src).toBe(fakeItem.image);
		expect(img.props().alt).toBe(fakeItem.title);
	});

	it('renders out the buttons', () => {
		const wrapper = shallow(<ItemComponent item={fakeItem} />);
		const buttonList = wrapper.find('.buttonList');

		expect(buttonList.children()).toHaveLength(3);
		expect(buttonList.find('Link')).toHaveLength(1);
		expect(buttonList.find('AddToCart')).toHaveLength(1);
		expect(buttonList.find('DeleteItem')).toHaveLength(1);
	});
});
