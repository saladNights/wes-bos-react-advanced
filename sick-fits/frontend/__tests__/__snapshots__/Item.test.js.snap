// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders and matches the snapshot 1`] = `
<ForwardRef(ItemStyles__Item)>
  <img
    alt="An Item"
    src="dog.jpg"
  />
  <ForwardRef(Title)>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "234rf2t",
          },
        }
      }
    >
      <a>
        An Item
      </a>
    </Link>
  </ForwardRef(Title)>
  <ForwardRef(PriceTag)>
    $50
  </ForwardRef(PriceTag)>
  <p>
    This is an item description
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "234rf2t",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="234rf2t"
    />
    <DeleteItem
      id="234rf2t"
    >
      Delete Item
    </DeleteItem>
  </div>
</ForwardRef(ItemStyles__Item)>
`;
